File: CS-Examples/08_FilteringAndValidation/CreateFilter.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

using Spire.Xls;
using Spire.Xls.Charts;

namespace CreateFilter
{
	public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, System.EventArgs e)
		{
            // Create a workbook to store Excel data
            Workbook workbook = new Workbook();

            // Load the Excel document from disk into the workbook
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\CreateFilter.xlsx");

            // Get the first worksheet in the workbook
            Worksheet sheet = workbook.Worksheets[0];

            // Specify the range for creating the filter (in this case, A1 to J1)
            sheet.AutoFilters.Range = sheet.Range["A1:J1"];

            // Save the modified workbook with the applied filter to a new file named "CreateFilter_out.xlsx"
            string result = "CreateFilter_out.xlsx";
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the file
            ExcelDocViewer(result);
		}

		private void ExcelDocViewer( string fileName )
		{
			try
			{
				System.Diagnostics.Process.Start(fileName);
			}
			catch{}
		}

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
	}
}

File: VB-Examples/08_FilteringAndValidation/CreateFilter.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel

Imports Spire.Xls
Imports Spire.Xls.Charts

Namespace CreateFilter
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object to represent an Excel workbook
            Dim workbook As New Workbook()

            ' Load an existing Excel file named "CreateFilter.xlsx" from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\CreateFilter.xlsx")

            ' Get the first worksheet (index 0) from the workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Set the range for auto-filtering to be from cell A1 to J1 in the worksheet
            sheet.AutoFilters.Range = sheet.Range("A1:J1")

            ' Specify the output filename for the filtered data
            Dim result As String = "CreateFilter_out.xlsx"

            ' Save the filtered data to a new Excel file with Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010)
            ' Release the resources used by the workbook
            workbook.Dispose()

            ExcelDocViewer(result)
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
